src/main/java/uk/gov/hmcts/reform/managecase/service/noc/RequestNoticeOfChangeService.java

```diff
@@ -108,6 +123,40 @@ private CaseDetails generateNoCRequestEvent(String caseId,
         return dataStoreRepository.submitNoticeOfChangeRequestEvent(caseId, eventId, changeOrganisationRequest);
     }
 
+    private DynamicList createCaseRoleIdDynamicList(String caseRoleId, String caseTypeId) {
+
+        Optional<CaseRole> caseRole = getCaseRolesDefinitions(caseRoleId, caseTypeId);
+
+        if (caseRole.isPresent()) {
+            DynamicListElement element = DynamicListElement.builder()
+                .code(caseRole.get().getId())
+                .label(caseRole.get().getName())
+                .build();
+            return DynamicList.builder()
+                .value(element)
+                .listItems(List.of(element))
+                .build();
+        }
+        return null;
+    }
+
+    private Optional<CaseRole> getCaseRolesDefinitions(String caseRole, String caseType) {
+        String caseRoleLowerCase = caseRole.toLowerCase();
+        // "0" specified for first two parameters as they are not used by definition store API
+        List<CaseRole> caseRolesDefinition = definitionStoreRepository.caseRoles("0", "0", caseType);
+        List<String> caseRolesDefinitionIds = caseRolesDefinition.stream()
+            .map(CaseRole::getId)
+            .map(String::toLowerCase)
+            .collect(toList());
+        if (!caseRolesDefinitionIds.contains(caseRoleLowerCase)) {
+            throw new ValidationException(
+                format(MISSING_COR_CASE_ROLE_ID_IN_CASE_DEFINITION, caseRole));
+        }
+
+        return caseRolesDefinition.stream()
```

Comment:
Can this be simplified like below 

`        return caseRolesDefinition.stream()
            .filter(cr -> caseRoleLowerCase.equalsIgnoreCase(cr.getId().toLowerCase()))
            .findFirst()
            .orElseThrow(() -> new ValidationException(
            format(MISSING_COR_CASE_ROLE_ID_IN_CASE_DEFINITION, caseRole)));`